src/Func.cpp

```diff
@@ -164,9 +164,10 @@ void Func::define_extern(const std::string &function_name,
                          const std::vector<Type> &types,
                          int dimensionality,
                          NameMangling mangling,
-                         bool uses_old_buffer_t) {
+                         bool uses_old_buffer_t,
+                         DeviceAPI device_api) {
```

Comment:
We had a discussion about this where we decided something along these lines:

uses_old_buffer_t should be an enum rather than a bool
define_extern should take some number of trailing ExternAttribute objects after the dimensionality parameter
You can construct an ExternAttribute from a DeviceAPI, NameMangling, or whatever the new old_buffer_t enum is.

That way order doesn't matter.